x/dsmr/node_test.go

```diff
@@ -1087,122 +1056,142 @@ func Test_Verify(t *testing.T) {
 func Test_Verify_BadBlock(t *testing.T) {
 	tests := []struct {
 		name    string
-		blk     func(chunkCert ChunkCertificate, parent Block) Block
+		blk     func(parent Block) Block
 		wantErr error
 	}{
 		{
 			name: "invalid parent",
-			blk: func(chunkCert ChunkCertificate, parent Block) Block {
+			blk: func(parent Block) Block {
 				return Block{
-					ParentID:   ids.GenerateTestID(),
-					Height:     parent.Height + 1,
-					Timestamp:  parent.Timestamp + 1,
-					ChunkCerts: []*ChunkCertificate{&chunkCert},
+					BlockHeader: BlockHeader{
+						ParentID:  ids.GenerateTestID(),
+						Height:    parent.Height + 1,
+						Timestamp: parent.Timestamp + 1,
+					},
+					ChunkCerts: parent.ChunkCerts,
```

Comment:
Going over this - I think this is either weird/or a bug that we allow duplicate chunk certs to pass verification. For now i'm doing this because it passes tests and it's convenient - but probably something to fix as a follow-up.

Reply:
It's a bug fixed in https://github.com/ava-labs/hypersdk/pull/1810. We should not allow duplicates either in a block or across the ancestry.